market/models.py

```diff
@@ -54,21 +53,28 @@ def get_absolute_url(self):
     def user_buy_stocks(self, quantity):
         if quantity <= self.stocks_remaining:
             self.stocks_remaining -= quantity
-            self.temp_stocks_bought += quantity
             self.save()
             return True
         return False
 
     def user_sell_stocks(self, quantity):
         if quantity <= self.stocks_offered:
             self.stocks_remaining += quantity
-            self.temp_stocks_sold += quantity
             self.save()
             return True
         return False
 
 
 def pre_save_company_receiver(sender, instance, *args, **kwargs):
+
+    # Setting the maximum stocks that a user can own for a company
+    if instance.cap_type == 'small':
+        instance.max_stocks_sell = float(instance.stocks_offered) * 0.18
```

Comment:
Is this line of code working? As the `max_stocks_sell` is an integer field and the value being given to it is a float.

Reply:
The casting of instance.stocks_offered to float was removed